ghmirror/core/mirror_requests.py

```diff
@@ -83,10 +108,28 @@ def online_request(method, url, auth, data=None):
         if last_mod is not None:
             headers['If-Modified-Since'] = last_mod
 
-    resp = requests.request(method=method, url=url, headers=headers,
-                            timeout=REQUESTS_TIMEOUT)
+    resp = requests.request(method=method,
+                            url=url,
+                            headers=headers,
+                            timeout=REQUESTS_TIMEOUT,
+                            params=parameters)
 
     if resp.status_code == 304:
+        if len(cached_response.json()) == per_page_elements and \
```

Comment:
can we move the cached_response content parsing logic into the `MirrorResponse` class?

Reply:
`cached_response` is not a MirrorResponse, is a "normal" request response, do you mean cast the response to be a MirrorResponse and then implement the logic there?

Reply:
You are right, I got confused about it. Forget I said anything here :)